src/helpers/getMovieUtils.ts

```diff
@@ -17,31 +17,51 @@ const getMoviesByName = async (name: string): Promise<MovieData> => {
  * This function is ran for specific <ShowCard /> data with a movieID.
  * @returns {Promise<MovieDetailsData>} | Specific data for a movie that is not originally supplied by getMoviesByName.
  */
-const getMovieDetails = async (id: number): Promise<MovieDetailsData> => {
+const getMovieDetails = async (id: number): Promise<ShowData> => {
     const response = await fetch(
         `https://api.themoviedb.org/3/movie/${id}?api_key=${
             import.meta.env.VITE_MOVIEDB_KEY
         }&append_to_response=images,release_dates`
     );
-    return response.json() as Promise<MovieDetailsData>;
+    const data = (await response.json()) as MovieDetailsData;
+    const returnRating = async (arr: MovieDetailsData) => {
```

Comment:
This does not need to be async.
```suggestion
    const returnRating = (arr: MovieDetailsData) => {
```